tests/test_client_functional.py

```diff
@@ -2453,6 +2458,53 @@ async def handler_redirect(request):
     assert data == body
 
 
+@pytest.mark.parametrize(
+    "invalid_redirect_url, error_message_url, expected_exception_class",
+    (
+        # yarl.URL.__new__ raises ValueError
+        ("http://:/", "http://:/", InvalidUrlRedirectClientError),
+        (
+            "http://example.org:non_int_port/",
+            "http://example.org:non_int_port/",
+            InvalidUrlRedirectClientError,
+        ),
+        # # yarl.URL.origin raises ValueError
+        ("http:/", "http:///", InvalidUrlRedirectClientError),
+        ("http:/example.com", "http:///example.com", InvalidUrlRedirectClientError),
+        ("http:///example.com", "http:///example.com", InvalidUrlRedirectClientError),
+        ("bluesky://d:i:d", "bluesky://d:i:d", InvalidUrlRedirectClientError),
+        ("call:+3801234567", r"call:\+3801234567", NonHttpUrlRedirectClientError),
+        ("skype:handle", "skype:handle", NonHttpUrlRedirectClientError),
+        (
+            "slack://instance/room",
+            "slack://instance/room",
+            NonHttpUrlRedirectClientError,
+        ),
+        ("steam:code", "steam:code", NonHttpUrlRedirectClientError),
+        ("twitter://handle", "twitter://handle", NonHttpUrlRedirectClientError),
+    ),
+)
+async def test_invalid_redirect_url(
```

Comment:
Per https://github.com/aio-libs/aiohttp/pull/6722#discussion_r1480006621 I want to see these clearly visible test scenarios:
* [ ] No HTTP handler present in the test, an exception is raised prior to making an HTTP request due to the initial input validation (the cases of a non-HTTP URL and an invalid URL)
* [ ] An HTTP handler is present in the test and it immediately redirects somewhere (non-HTTP URL and invalid URL scenarios)
* [ ] Several HTTP handlers are present in the test. The first one redirects to the second one through HTTP, without any problems, the second one redirects to the third one. And only the fourth handler redirects to something exceptional (non-HTTP URL and invalid URL scenarios)

Reply:
I added new tests - please take a look and mark tasks as resolved if are ok.

Reply:
Marking this as resolved. Pending the suggested fixes.